### PR TITLE
Add a rule to ease the migration to v0.9.28

### PR DESCRIPTION
**Pull request by mlachkar**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,10 +7,10 @@ jobs:
...
       - run: ./scalafmt --test
-  build:
+  migration-rules:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: olafurpg/setup-scala@v10
       - run: sbt test
-      - run: cd scalafix && sbt test
+      - run: cd migration-rules && sbt test
```
Comment (bjaglin) [inline .github/workflows/ci.yml:16]: I think there is something off in the layout, https://github.com/scalacenter/scalafix.g8/tree/9073804c25967bb94dc2b13c75d3d42da724a753/migration-rules should only contain 2 directories, and this should iterate over them

**Review by bjaglin — COMMENTED** [review]
That looks promising I think!

I wonder if/how the previous migration was referenced - the instructions in https://github.com/scalacenter/scalafix.g8/blob/9073804c25967bb94dc2b13c75d3d42da724a753/migration-rules/readme.md don't look like they are working. It's not on steward either, so I guess it's OK as it's very old anyway.

**Comment by mlachkar** [comment]
I will put the old rule under it's folder, and delete/fix the readme

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,10 +7,16 @@ jobs:
...
       - uses: olafurpg/setup-scala@v10
       - run: sbt test
-      - run: cd scalafix && sbt test
+  migration-rules:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v10
+      - run: cd migration-rules/v0.9.28 && sbt test
+      - run: cd migration-rules/v0.6.x && sbt test
```
Comment (bjaglin) [inline .github/workflows/ci.yml:22]: would this work?
```suggestion
      - run: find migration-rules/ -mindepth 1 -maxdepth 1 -type d -execdir sh -c 'cd {} && sbt test' \;
```

**Review by bjaglin — COMMENTED** [review]
you missed moving 2 files at the root https://github.com/scalacenter/scalafix.g8/tree/7672ae7c3880822c26a44f2853c108593467c1be/migration-rules

File: readme.md [hunk 3]
```diff
@@ -19,11 +19,14 @@ For example, if the rule contains migration rules for version
 If the rule is a linter that prohibits vars, the name can be `NoVars`.
 
 
-## Scalafix v0.6 migration
+## migration-rules
```
Comment (bjaglin) [inline readme.md:22]: :+1:

File: migration-rules/v0.9.28/tests/src/test/scala/fix/RuleSuite.scala [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package fix
+
+import scalafix.testkit.AbstractSemanticRuleSuite
+import org.scalatest.FunSuiteLike
+
+class wRuleSuite extends AbstractSemanticRuleSuite with FunSuiteLike {
```
Comment (bjaglin) [inline migration-rules/v0.9.28/tests/src/test/scala/fix/RuleSuite.scala:6]: not that it matters, but it caught my attention
```suggestion
class RuleSuite extends AbstractSemanticRuleSuite with FunSuiteLike {
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -7,10 +7,15 @@ jobs:
...
       - uses: actions/checkout@v2
       - uses: olafurpg/setup-scala@v10
       - run: sbt test
-      - run: cd scalafix && sbt test
+  migration-rules:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v10
+      - run: find migration-rules/ -mindepth 1 -maxdepth 1 -type d -execdir sh -c 'cd {} && sbt test' \;
```
Comment (bjaglin) [inline .github/workflows/ci.yml:21]: wow, it's working! https://github.com/scalacenter/scalafix.g8/pull/20/checks?check_run_id=2602704306

**Review by bjaglin — APPROVED** [review]
great preparation for future upgrade to better support rule authors!

**Pull request merged**